### DISCUSSION TITLE
Is it possible to use colors from pywal in the promt theme?

### DISCUSSION THREAD

**Discussion by mylinuxforwork** [JanDeDobbeleer/oh-my-posh#6010] [General]
Hi all,

I want to use colors from pywal generated from the wallpaper to color some parts of my promt. Is there a way to include a color palette file into the promt theme configuration and use the variables?

thanks in advance for your ideas.

Stephan

**Comment by JanDeDobbeleer** [comment]
@mylinuxforwork you can have a look at the `palette` functionality, but that requires you to add it manually. See [here](https://www.ohmyposh.dev/docs/configuration/colors#palette).

**Reply by mylinuxforwork** [reply]
Thank you. Cool would be if it would be possible to include/source a palette conf file into the configuration. The palette file could be dynamically generated by pywal.

**Reply by tom-gora** [reply]
it is absolutely possible with just a little of scripting. Not sure if there are easy tools to do field manipulation if you have toml or yaml config but with json JQ makes this dead simple. Prepare yourself a template file that corresponds to your "palette" object. I.E. I use wallust and I have a wallust_template_omp.json file like this:
```json
{
  "urgent": "{{color13 | saturate(0.2) }}",
  "orange": "{{color4 | saturate(0.2)}}",
  "os": "{{color15 | lighten(0.2) | saturate(0.2)}}",
  "purple": "{{color9 | saturate(0.4)}}",
  "transient": "{{foreground | darken(0.7)}}",
  "user": "p:os"
}
```
that's it. Just an object, no array.
Then in my environment updating script run after I got a new palette extracted and templates written I placed:
```bash
	# handle oh-my-posh
	local WALLUST_OMP_PALETTE_FILE="$HOME/.config/wallust/targets/wallust_omp.json"
	local OMP_CONFIG_FILE="$HOME/.config/my_omp.json"

	if [[ ! -f "$WALLUST_OMP_PALETTE_FILE" ]]; then
		echo "Error: New Oh My Posh palette file not found at $WALLUST_OMP_PALETTE_FILE." >&2
		return 1
	fi

	if [[ ! -f "$OMP_CONFIG_FILE" ]]; then
		echo "Error: Oh My Posh config file not found at $OMP_CONFIG_FILE." >&2
		return 1
	fi

	local NEW_PALETTE_CONTENT NEW_CONFIG_CONTENT
	NEW_PALETTE_CONTENT=$(<"$WALLUST_OMP_PALETTE_FILE")

	NEW_CONFIG_CONTENT=$(jq --argjson new_palette "$NEW_PALETTE_CONTENT" '.palette = $new_palette' "$OMP_CONFIG_FILE")
	echo "$NEW_CONFIG_CONTENT" >"$OMP_CONFIG_FILE"
	if [ $? -ne 0 ]; then
		echo "Error: Failed to update Oh My Posh config file with jq." >&2
		return 1
	fi
  #...
  ```

Basically, read the templated "palette" bit, swap this part in the config omp config and write. Then you're done. When you press enter a prompt with new colors appears.  
EDIT: Recording


https://github.com/user-attachments/assets/66677131-8271-4c85-b14d-4572bf44f2e4
